src/cdk/stepper/stepper.ts

```diff
@@ -283,15 +349,48 @@ export class CdkStepper implements AfterViewInit, OnDestroy {
   }
 
   /** Returns the type of icon to be displayed. */
-  _getIndicatorType(index: number): 'number' | 'edit' | 'done' {
+  _getIndicatorType(index: number, state: StepState = STEP_STATE.NUMBER): StepState {
     const step = this._steps.toArray()[index];
-    if (!step.completed || this._selectedIndex == index) {
-      return 'number';
+    const isCurrentStep = this._isCurrentStep(index);
+
+    if (step._useGuidelines) {
+      return this._getGuidelineLogic(step, isCurrentStep, state);
     } else {
-      return step.editable ? 'edit' : 'done';
+      return this._getDefaultIndicatorLogic(step, isCurrentStep);
     }
   }
 
+  private _getDefaultIndicatorLogic(step: CdkStep, isCurrentStep: boolean): StepState {
+    if (step._showError && step.hasError && !isCurrentStep) {
+      return STEP_STATE.ERROR;
+    } else if (!step.completed || isCurrentStep) {
+      return STEP_STATE.NUMBER;
+    } else {
+      return step.editable ? STEP_STATE.EDIT : STEP_STATE.DONE;
+    }
+  }
+
+  private _getGuidelineLogic(
+    step: CdkStep,
+    isCurrentStep: boolean,
+    state: StepState = STEP_STATE.NUMBER): StepState {
+    if (step._showError && step.hasError && !isCurrentStep) {
+      return STEP_STATE.ERROR;
```

Comment:
It's still nice to have, don't you agree? If for some reason we decide to change the `ERROR` string to something else, we would only need to change it in one place. Let me know, thanks!

Reply:
I guess it's more of a preference since the enum would end up generating a bit more runtime code and the values aren't likely to change. I'm fine with keeping it as it is.